### ISSUE TITLE
About reproducibility of results on ModelNet40 and ModelNet10

### ISSUE THREAD

**Issue by Azx030512** [qimaqi/ShapeSplat-Gaussian_MAE#14]
Hello, thanks for your work.
I use your official code to pretrain the Gaussian-MAE model and fine-tune it on ModelNet40 and ModelNet10 datasets with gaussian version. But only  90.75% accuracy on ModelNet40 and 93.28% accuracy on ModelNet10 were obtained with three runs, which is about 2% lower than the result reported in your 3DV paper.
More importantly, the scale-and-translate data augmentation can not be triggered with this `if ["xyz"] not in attribute:` but should be `if "xyz" not in attribute:`, I think this augmentation is important for reproduce the results. But even I fixed it by myself, the results reproduced are still 2% lower than reported in the paper. 
Can you explain the potential reason or provide the checkpoints? Many thanks!
![Image](https://github.com/user-attachments/assets/bac996f9-ffad-4e97-b774-72090169d743)

**Comment by willxxy** [comment]
We also have a problem with this. Any updates? @Azx030512 @qimaqi

**Comment by Azx030512** [comment]
No progress......

**Comment by qimaqi** [comment]
Dear Azx030512 and willxxy,

Thank you for your interest and patience. We are currently working towards a deadline and apologize for the delayed response. Could you please share the config files you used so that we can reproduce the issue and investigate it more effectively?

Best regards,
Qi

**Comment by Azx030512** [comment]
Sorry to bother you at this moment. However, simply using the default configuration files from your repository and following the provided instructions naturally does not reproduce the results reported in the paper.
Would it be possible for you to release the checkpoint files? For a pre-training model, releasing the pre-trained backbone is essential, as the model has limited value without it.

**Comment by YangParky** [comment]
> Sorry to bother you at this moment. However, simply using the default configuration files from your repository and following the provided instructions naturally does not reproduce the results reported in the paper. Would it be possible for you to release the checkpoint files? For a pre-training model, releasing the pre-trained backbone is essential, as the model has limited value without it.

Hi！I have the same issue and obtain the 90.76% acc at the 261-th epoch. May I ask you solve it now ?

**Comment by YangParky** [comment]
> Any updates?

 Any updates?

**Comment by qimaqi** [comment]
Dear YangParky

We're currently working on this and reproducing the results. We'll get back to you as soon as possible

Best
Qi

**Comment by YangParky** [comment]
Any updates?

**Comment by unique1i** [comment]
Hi all, thanks for this suggested issue! 

There are several bugs fixed in the configs and the training/finetuning process, including the `PointcloudScaleAndTranslate` augmentation used. One important config we didn't make clear before is that in the reported results tables, the Gaussian-MAE pretraining process using input 3DGS objects of 1k Gaussians, and classification performance on modelnet10 and modelnet40 is obtained using input of 4k Gaussians. Finetuning on objects of 1k Gaussians indeed leads to worse results. Now the finetuning [configs](https://github.com/qimaqi/ShapeSplat-Gaussian_MAE/tree/main/cfgs/finetune) and [scirpts](https://github.com/qimaqi/ShapeSplat-Gaussian_MAE/tree/main/sh_jobs/finetune) are updated to address this.

We reproduce the modelnet10/modelnet40 finetuning classification results using the current codebase in the following table, which are consistent with the results reported in our paper. The best results are obtained with pretraining using objects of 1k Gaussians and finetuning on 4k Gaussians. Note the results will vary during different runs.

| GS number | pretrain 1k |  | pretrain 4k |  |
|--------|-------------|-------------|-------------|-------------|
| **soft_knn** | True | False | True | False |
| **finetune 4k** | **95.70484**/<br>92.41994 | 95.37445/<br>**93.43332** | 95.48458/<br>92.41994 | 95.26431/<br>93.06851 |

The model checkpoints are updated at [gaussian_mae_ckpts](https://huggingface.co/datasets/ShapeSplats/sharing/tree/main/gaussian_mae_ckpts). The corresponding wandb training logs of the reproduced runs are shared here [wandb_logs](https://api.wandb.ai/links/uniqueli/gs81sy66).

The configuration we used will be made clear in the arXiv version, and the inconsistency in the main paper ablation Table 5 will be updated. I will post a follow-up comment once it's done.

Again, we thank you for raising this point and for helping to improve the repo. Please let me know if any other questions.

**Comment by Azx030512** [comment]
Many thanks for your reply!

**Comment by unique1i** [comment]
The [arxiv version](https://arxiv.org/abs/2408.10906) is now updated to include this config we used for the main results.
